qlf_fasm/qlf_fasm.py

```diff
@@ -0,0 +1,513 @@
+#!/usr/bin/env python3
+"""
+QuickLogic qlf-series FPGA FASM to bitstream and bitstream to FASM conversion
+utility.
+"""
+import argparse
+import os
+import re
+import logging
+import pkg_resources
+
+import fasm
+
+from .database import Bit, Database
+from .bitstream import TextBitstream
+
+# =============================================================================
+
+DATABASES = {
+    "qlf_k4n8": pkg_resources.resource_filename("qlf_fasm", "database/qlf_k4n8")
+}
+
+# =============================================================================
+
+
+class QlfFasmAssembler():
+    """
+    FASM assembler for QuickLogic QLF devices.
+    """
+
+    LOC_RE = re.compile(r"(?P<name>.+)_(?P<x>[0-9]+)__(?P<y>[0-9]+)_$")
```

Comment:
Worth adding a comment with an example of a matched string

Reply:
Comment + examples added.